File: mobile-sdk/android/ExampleIntegration.kt
/**
 * FMCWS Android SDK — пример интеграции в банковское приложение
 * Файл: ExampleIntegration.kt
 */

package by.fmcws.example

import android.app.Application
import android.app.AlertDialog
import android.content.Context
import by.fmcws.sdk.*

// ── 1. Инициализация в Application ────────────────────────────────────────────
class BankApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        FmcwsSDK.init(
            appContext = this,
            cfg = FmcwsConfig(
                apiUrl   = "https://fmcws.bank.by",
                apiKey   = BuildConfig.FMCWS_API_KEY,   // из gradle.properties
                clientId = "anonymous",                  // обновляется после логина
                debug    = BuildConfig.DEBUG,
            )
        )
    }
}

// ── 2. Реакция на предупреждения ───────────────────────────────────────────────
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Регистрируем listener
        FmcwsSDK.setWarningListener(object : FmcwsWarningListener {
            override fun onWarning(warning: FmcwsWarning) {
                showFraudWarningDialog(warning)
            }
        })
    }

    private fun showFraudWarningDialog(warning: FmcwsWarning) {
        val (title, icon) = when (warning.level) {
            FmcwsRiskLevel.CRITICAL -> "Критическое предупреждение" to "🚨"
            FmcwsRiskLevel.HIGH     -> "Подозрительная активность"  to "🔶"
            FmcwsRiskLevel.MEDIUM   -> "Внимание"                   to "⚠️"
            FmcwsRiskLevel.LOW      -> return   // LOW не показываем
        }
        AlertDialog.Builder(this)
            .setTitle("$icon $title")
            .setMessage(warning.message)
            .setPositiveButton("Понятно") { d, _ -> d.dismiss() }
            .apply {
                if (warning.level == FmcwsRiskLevel.CRITICAL) {
                    setNeutralButton("Позвонить в банк") { _, _ ->
                        // startActivity(Intent(Intent.ACTION_DIAL, Uri.parse("tel:+375XXXXXXXXX")))
                    }
                }
            }
            .setCancelable(warning.level != FmcwsRiskLevel.CRITICAL)
            .show()
    }
}

// ── 3. Трекинг событий в нужных местах ────────────────────────────────────────
class AuthRepository {
    fun onLoginSuccess(pseudoClientId: String, isNewDevice: Boolean) {
        FmcwsSDK.identify(pseudoClientId)
        FmcwsSDK.track("LOGIN_SUCCESS", mapOf(
            "is_new_device" to isNewDevice,
        ))
    }
    fun onLoginFail() = FmcwsSDK.track("LOGIN_FAIL")
    fun onLogout()    { FmcwsSDK.reset() }
}

class TransferRepository {
    fun onTransferCreate(amount: Double, currency: String, isNewRecipient: Boolean) {
        FmcwsSDK.track("TRANSFER_CREATE", mapOf(
            "amount"          to amount,
            "currency"        to currency,
            "is_new_recipient" to isNewRecipient,
        ))
    }
    fun onTransferConfirm(transferId: String) {
        FmcwsSDK.track("TRANSFER_CONFIRM", mapOf("transfer_id" to transferId))
    }
    fun onRecipientAdd(recipientHash: String) {
        FmcwsSDK.track("RECIPIENT_ADD", mapOf("recipient_hash" to recipientHash))
    }
}

class ProfileRepository {
    fun onPhoneChange()  = FmcwsSDK.track("CONTACT_CHANGE", mapOf("field" to "phone"))
    fun onEmailChange()  = FmcwsSDK.track("CONTACT_CHANGE", mapOf("field" to "email"))
    fun onLimitChange()  = FmcwsSDK.track("LIMIT_CHANGE")
    fun onOtpRequest()   = FmcwsSDK.track("OTP_REQUEST")
    fun onOtpConfirm()   = FmcwsSDK.track("OTP_CONFIRM")
    fun onVirtualCardIssue() = FmcwsSDK.track("VIRTUAL_CARD_ISSUE")
}

File: mobile-sdk/ios/ExampleIntegration.swift
/**
 * FMCWS iOS SDK — пример интеграции в банковское приложение
 * Файл: ExampleIntegration.swift
 */

import SwiftUI
import FMCWS

// ── 1. Инициализация (SwiftUI @main или AppDelegate) ─────────────────────────
@main
struct BankApp: App {
    init() {
        FmcwsSDK.shared.initialize(
            config: FmcwsConfig(
                apiURL:   URL(string: "https://fmcws.bank.by")!,
                apiKey:   Secrets.fmcwsApiKey,         // из .xcconfig / Keychain
                clientId: "anonymous",                  // обновляется после логина
                debug:    false
            )
        )
    }
    var body: some Scene { WindowGroup { RootView() } }
}

// ── 2. Реакция на предупреждения ───────────────────────────────────────────────
class AppDelegate: NSObject, UIApplicationDelegate, FmcwsWarningDelegate {
    func application(_ application: UIApplication,
                     didFinishLaunchingWithOptions _: [UIApplication.LaunchOptionsKey : Any]? = nil) -> Bool {
        FmcwsSDK.shared.warningDelegate = self
        return true
    }

    func fmcws(_ sdk: FmcwsSDK, didReceiveWarning warning: FmcwsWarning) {
        // Находим активный UIViewController и показываем alert
        guard let vc = UIApplication.shared.connectedScenes
            .compactMap({ ($0 as? UIWindowScene)?.keyWindow?.rootViewController })
            .first else { return }

        let (title, icon): (String, String)
        switch warning.level {
        case .critical: (title, icon) = ("Критическое предупреждение", "🚨")
        case .high:     (title, icon) = ("Подозрительная активность",  "🔶")
        case .medium:   (title, icon) = ("Внимание",                   "⚠️")
        case .low:      return
        }

        let alert = UIAlertController(
            title:   "\(icon) \(title)",
            message: warning.message,
            preferredStyle: .alert
        )
        alert.addAction(UIAlertAction(title: "Понятно", style: .default))
        if warning.level == .critical {
            alert.addAction(UIAlertAction(title: "Позвонить в банк", style: .destructive) { _ in
                // UIApplication.shared.open(URL(string: "tel:+375XXXXXXXXX")!)
            })
        }
        vc.present(alert, animated: true)
    }
}

// ── 3. Трекинг событий ────────────────────────────────────────────────────────
struct AuthService {
    func onLoginSuccess(pseudoClientId: String, isNewDevice: Bool) {
        FmcwsSDK.shared.identify(pseudoClientId)
        FmcwsSDK.shared.track("LOGIN_SUCCESS", extra: [
            "is_new_device": isNewDevice
        ])
    }
    func onLoginFail() { FmcwsSDK.shared.track("LOGIN_FAIL") }
    func onLogout()    { FmcwsSDK.shared.reset() }
}

struct TransferService {
    func onTransferCreate(amount: Double, currency: String, isNewRecipient: Bool) {
        FmcwsSDK.shared.track("TRANSFER_CREATE", extra: [
            "amount":           amount,
            "currency":         currency,
            "is_new_recipient": isNewRecipient,
        ])
    }
    func onTransferConfirm(transferId: String) {
        FmcwsSDK.shared.track("TRANSFER_CONFIRM", extra: ["transfer_id": transferId])
    }
    func onRecipientAdd(recipientHash: String) {
        FmcwsSDK.shared.track("RECIPIENT_ADD", extra: ["recipient_hash": recipientHash])
    }
}

struct ProfileService {
    func onPhoneChange()     { FmcwsSDK.shared.track("CONTACT_CHANGE", extra: ["field": "phone"]) }
    func onEmailChange()     { FmcwsSDK.shared.track("CONTACT_CHANGE", extra: ["field": "email"]) }
    func onLimitChange()     { FmcwsSDK.shared.track("LIMIT_CHANGE") }
    func onOtpRequest()      { FmcwsSDK.shared.track("OTP_REQUEST") }
    func onVirtualCardIssue(){ FmcwsSDK.shared.track("VIRTUAL_CARD_ISSUE") }
}
